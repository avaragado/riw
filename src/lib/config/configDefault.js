// @flow

const cfg: RIWConfig = {
    configFile: null, // overridden
    rootDir: '', // overridden
    defaultLocale: 'en-us',
    targetLocales: [],
    translationsDatabaseFile: 'src/locale/riw-db.json',
    sourceDirs: ['src/**/*.js'],
    collateDir: 'tmp/babel-plugin-react-intl',
    inputMode: 'source',
    translationsOutputFile: 'src/locale/[locale].json',
    outputMode: 'file-per-locale',
    todoFile: 'src/locale/TODO-untranslated.json',
};

export default cfg;
