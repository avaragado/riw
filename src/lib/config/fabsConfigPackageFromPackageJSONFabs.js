// @flow

import path from 'path';

export default (fabsPackageJSON: ?AbsolutePath): ?AbsolutePath => (
    fabsPackageJSON
        ? path.resolve(path.dirname(fabsPackageJSON), '.riw-config.js')
        : null
    );
