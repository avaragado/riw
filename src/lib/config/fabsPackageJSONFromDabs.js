// @flow

import findUp from 'find-up';

export default (cwd: AbsolutePath): ?AbsolutePath => findUp.sync('package.json', { cwd });

