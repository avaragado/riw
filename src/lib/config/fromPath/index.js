// @flow

import path from 'path';
import chalk from 'chalk';

import type { Path, AbsolutePath } from '../../../types';
import type { ConfigSparse, ConfigSparseWithSource } from '../../config';
import log from '../../log';

export default (pathConfig: Path): ?ConfigSparseWithSource => {
    const fabs: AbsolutePath = path.resolve(pathConfig);

    try {
        return {
            ...(require(fabs): ConfigSparse),
            configFile: fabs,
        };

    } catch (err) {
        log.error('File not found: ', chalk.bold(fabs));
        return null;
    }
};
