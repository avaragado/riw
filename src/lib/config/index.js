// @flow

import configDefault from './resolve/configDefault';
import fabsConfigPackageForCWD from './fromPackage/fabsConfigPackageForCWD';
import configFromPath from './fromPath';
import configFromPackage from './fromPackage';
import configResolve from './resolve';

export {
    configDefault,
    fabsConfigPackageForCWD,
    configFromPath,
    configFromPackage,
    configResolve,
};
