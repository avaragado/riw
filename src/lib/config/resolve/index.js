// @flow

import path from 'path';

import type { Config, ConfigSparseWithSource } from '../../config';

import configDefault from './configDefault';

const maybeAddRootDir = ({ configFile, rootDir }) => (
    configFile && !rootDir
        ? { rootDir: path.dirname(configFile) }
        : {}
);

export default (config: ConfigSparseWithSource): Config => ({
    ...configDefault,
    ...config,
    ...maybeAddRootDir(config),
});
