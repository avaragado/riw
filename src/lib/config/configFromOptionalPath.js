// @flow

import path from 'path';

import log from '../log';

import configDefault from './configDefault';
import fabsPackageJSONFromDabs from './fabsPackageJSONFromDabs';
import fabsConfigPackageFromPackageJSONFabs from './fabsConfigPackageFromPackageJSONFabs';

const requireOrNull = (fabs: AbsolutePath): ?Object => {
    try {
        return require(fabs);

    } catch (err) {
        return null;
    }
};

const configFromPath = (pathConfig: Path): ?RIWConfig => {
    const fabs: AbsolutePath = path.resolve(pathConfig);
    const configOverride = requireOrNull(fabs);

    if (configOverride === null) {
        log.error('riw', 'File not found: %j.', fabs);
        return null;
    }

    return {
        ...configDefault,
        ...configOverride,
        rootDir: path.dirname(fabs),
    };
};

const configFromPackage = (): ?RIWConfig => {
    const fabsPackageJSON = fabsPackageJSONFromDabs(process.cwd());
    const fabsConfigPackage = fabsConfigPackageFromPackageJSONFabs(fabsPackageJSON);

    if (!fabsConfigPackage || !fabsPackageJSON) {
        log.error('riw', 'Unable to find package root');
        return null;
    }

    let configOverride: ?Object = requireOrNull(fabsConfigPackage);

    if (configOverride === null) {
        configOverride = requireOrNull(fabsPackageJSON);

        if (configOverride && configOverride.riw) {
            configOverride = configOverride.riw;
        } else {
            configOverride = null;
        }
    }

    if (configOverride === null) {
        log.warn('riw', 'No config located for the project. You probably want to add some configuration.');
        configOverride = {};
    }

    return {
        ...configDefault,
        ...configOverride,
        rootDir: path.dirname(fabsConfigPackage),
    };

};

export default (pathConfig: ?Path): ?RIWConfig => (
    pathConfig
        ? configFromPath(pathConfig)
        : configFromPackage()
);
