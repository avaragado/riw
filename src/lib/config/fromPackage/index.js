// @flow

import log from '../../log';

import fabsPackageJSONFromDabs from './fabsPackageJSONFromDabs';
import fabsConfigPackageFromPackageJSONFabs from './fabsConfigPackageFromPackageJSONFabs';

export default (): ?RIWConfigSparseWithSource => {
    const fabsPackageJSON = fabsPackageJSONFromDabs(process.cwd());
    const fabsConfigPackage = fabsConfigPackageFromPackageJSONFabs(fabsPackageJSON);

    if (!fabsPackageJSON || !fabsConfigPackage) {
        log.error('riw', 'Unable to find package root');
        return null;
    }

    try {
        return {
            ...(require(fabsConfigPackage): RIWConfigSparse),
            configFile: fabsConfigPackage,
        };

    } catch (err) {
        // fabsPackageJSON exists as we hunted the fs for it above:
        // but it might not have the riw key.
        const config: RIWConfigSparse = require(fabsPackageJSON).riw;

        if (!config) {
            log.warn('riw', 'No app config found. You probably want to add some configuration.');
        }

        return {
            ...config,
            configFile: fabsPackageJSON,
        };
    }
};
