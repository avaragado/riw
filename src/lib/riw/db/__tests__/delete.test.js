// @flow

import fs from 'fs';

import mock from 'mock-fs';

import type { ConfigSparseWithSource } from '../../../config';
import { configResolve } from '../../../config';

import type { TranslationsDB, DBListSpec } from '../';
import del from '../delete';

const frelDB = 'db.json';

const cfgBase: ConfigSparseWithSource = {
    translationsDatabaseFile: frelDB,
};

const db: TranslationsDB = {
    version: 1,
    data: {
        'one': {
            'desc1': {
                'aa-aa': '[aa-aa]1 one',
                'bb-bb': '[bb-bb]1 one',
            },
            'desc2': {
                'aa-aa': '[aa-aa]2 one',
                'bb-bb': '[bb-bb]2 one',
            },
        },
        'two': {
            'desc1': {
                'aa-aa': '[aa-aa]1 two',
                'bb-bb': '[bb-bb]1 two',
            },
        },
        'three': {
            'desc1': {
                'aa-aa': '[aa-aa]1 three',
                'cc-cc': '[cc-cc]1 three',
            },
        },
    },
};

const jsonDB = JSON.stringify(db, null, 4);

type Fixture = {
    name: string,
    opt: DBListSpec,
};

const fixtures: Fixture[] = [
    {
        name: '01 defaultMessage match',
        opt: {
            match: {
                defaultMessage: 'two',
            },
        },
    },

    {
        name: '02 description match',
        opt: {
            match: {
                description: 'desc1',
            },
        },
    },

    {
        name: '03 locale match',
        opt: {
            match: {
                locale: 'bb-bb',
            },
        },
    },

    {
        name: '04 translation match',
        opt: {
            match: {
                translation: '[cc-cc]1 three',
            },
        },
    },
];

describe('lib/riw/db/delete', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            mock({
                [frelDB]: jsonDB,
            });

            const cfg = configResolve(cfgBase);

            del(cfg)(fixture.opt);

            const dbOut: TranslationsDB = JSON.parse(fs.readFileSync(frelDB).toString());

            mock.restore();

            expect(dbOut).toMatchSnapshot();
        });
    });
});
