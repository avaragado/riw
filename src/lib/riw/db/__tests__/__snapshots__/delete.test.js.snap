// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/riw/db/delete 01 defaultMessage match 1`] = `
Object {
  "data": Object {
    "one": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 one",
        "bb-bb": "[bb-bb]1 one",
      },
      "desc2": Object {
        "aa-aa": "[aa-aa]2 one",
        "bb-bb": "[bb-bb]2 one",
      },
    },
    "three": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 three",
        "cc-cc": "[cc-cc]1 three",
      },
    },
  },
  "version": 1,
}
`;

exports[`lib/riw/db/delete 02 description match 1`] = `
Object {
  "data": Object {
    "one": Object {
      "desc2": Object {
        "aa-aa": "[aa-aa]2 one",
        "bb-bb": "[bb-bb]2 one",
      },
    },
  },
  "version": 1,
}
`;

exports[`lib/riw/db/delete 03 locale match 1`] = `
Object {
  "data": Object {
    "one": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 one",
      },
      "desc2": Object {
        "aa-aa": "[aa-aa]2 one",
      },
    },
    "three": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 three",
        "cc-cc": "[cc-cc]1 three",
      },
    },
    "two": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 two",
      },
    },
  },
  "version": 1,
}
`;

exports[`lib/riw/db/delete 04 translation match 1`] = `
Object {
  "data": Object {
    "one": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 one",
        "bb-bb": "[bb-bb]1 one",
      },
      "desc2": Object {
        "aa-aa": "[aa-aa]2 one",
        "bb-bb": "[bb-bb]2 one",
      },
    },
    "three": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 three",
      },
    },
    "two": Object {
      "desc1": Object {
        "aa-aa": "[aa-aa]1 two",
        "bb-bb": "[bb-bb]1 two",
      },
    },
  },
  "version": 1,
}
`;
